### PR TITLE
[MM-38103] Include professional and enterprise SKUs in license check

### PR DESCRIPTION
**Pull request by mickmister**
#### Summary

Similar to mscalendar license change https://github.com/mattermost/mattermost-plugin-mscalendar/pull/223

#### Ticket Link

Fixes https://mattermost.atlassian.net/browse/MM-38103

### CODE REVIEW

**Review by larkox — APPROVED** [review]

**Review by hanzei — COMMENTED** [review]
https://pkg.go.dev/github.com/mattermost/mattermost-plugin-api#IsE20LicensedOrDevelopment got updated recently. Should this PR should make use of it?

File: server/license.go [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+)
+
+const (
+	e20          = "E20"
+	professional = "professional"
+	enterprise   = "enterprise"
+)
+
+func HasEnterpriseFeatures(config *model.Config, license *model.License) bool {
+	if license != nil && (license.SkuShortName == e20 || license.SkuShortName == enterprise || license.SkuShortName == professional) {
```
Comment (hanzei) [inline server/license.go:15]: Just to double check: The plugin should be able to run with Professional?

Comment (mickmister) [reply server/license.go:15]: @hanzei Yes this is correct.

> https://pkg.go.dev/github.com/mattermost/mattermost-plugin-api#IsE20LicensedOrDevelopment got updated recently. Should this PR should make use of it?

That function can't be used for this reason, there is no check for Professional in that function.

**Review by hanzei — APPROVED** [review]
Thanks for the clarification :+1:

**Comment by DHaussermann** [comment]
@mickmister I see a plugin crash once I have the plugin configured and enabled. This occurred when running the `start` command to post a connection link. 

Are you able to reproduce this?
```
{"timestamp":"2021-09-13 17:55:48.795 -04:00","level":"debug","msg":"Websocket request","caller":"wsapi/websocket_handler.go:25","action":"user_typing"}
{"timestamp":"2021-09-13 17:55:49.629 -04:00","level":"debug","msg":"panic: runtime error: invalid memory address or nil pointer dereference","caller":"plugin/hclog_adapter.go:54","plugin_id":"com.mattermost.msteamsmeetings"}
{"timestamp":"2021-09-13 17:55:49.629 -04:00","level":"debug","msg":"[signal SIGSEGV: segmentation violation code=0x1 addr=0x48 pc=0x16c29e6]","caller":"plugin/hclog_adapter.go:54","plugin_id":"com.mattermost.msteamsmeetings"}
{"timestamp":"2021-09-13 17:55:49.629 -04:00","level":"debug","msg":"","caller":"plugin/hclog_adapter.go:54","plugin_id":"com.mattermost.msteamsmeetings"}
{"timestamp":"2021-09-13 17:55:49.629 -04:00","level":"debug","msg":"goroutine 59 [running]:","caller":"plugin/hclog_adapter.go:54","plugin_id":"com.mattermost.msteamsmeetings"}
{"timestamp":"2021-09-13 17:55:49.629 -04:00","level":"debug","msg":"github.com/mattermost/mattermost-server/v5/model.(*AppError).Error(0x0, 0x29, 0x1b018e0)","caller":"plugin/hclog_adapter.go:54","plugin_id":"com.mattermost.msteamsmeetings"}
{"timestamp":"2021-09-13 17:55:49.629 -04:00","level":"debug","msg":"\t/Users/dylanhaussermann/go/pkg/mod/github.com/mattermost/mattermost-server/v5@v5.28.1/model/utils.go:73 +0x26","caller":"plugin/hclog_adapter.go:54","plugin_id":"com.mattermost.msteamsmeetings"}
{"timestamp":"2021-09-13 17:55:49.629 -04:00","level":"debug","msg":"github.com/pkg/errors.(*withMessage).Error(0xc0005c4b00, 0xc000511b48, 0x100bfec)","caller":"plugin/hclog_adapter.go:54","plugin_id":"com.mattermost.msteamsmeetings"}
{"timestamp":"2021-09-13 17:55:49.629 -04:00","level":"debug","msg":"\t/Users/dylanhaussermann/go/pkg/mod/github.com/pkg/errors@v0.9.1/errors.go:244 +0x37","caller":"plugin/hclog_adapter.go:54","plugin_id":"com.mattermost.msteamsmeetings"}
{"timestamp":"2021-09-13 17:55:49.629 -04:00","level":"debug","msg":"main.(*Plugin).ExecuteCommand(0xc000283b00, 0xc0003006c0, 0xc0003fe000, 0x2a0f6a18, 0xc000283b00)","caller":"plugin/hclog_adapter.go:54","plugin_id":"com.mattermost.msteamsmeetings"}
{"timestamp":"2021-09-13 17:55:49.629 -04:00","level":"debug","msg":"\t/Users/dylanhaussermann/go/src/github.com/mattermost/mattermost-plugin-msteams-meetings/server/command.go:129 +0xd3","caller":"plugin/hclog_adapter.go:54","plugin_id":"com.mattermost.msteamsmeetings"}
{"timestamp":"2021-09-13 17:55:49.629 -04:00","level":"debug","msg":"github.com/mattermost/mattermost-server/v5/plugin.(*hooksRPCServer).ExecuteCommand(0xc0000de960, 0xc0000b0710, 0xc0002741a0, 0x0, 0x0)","caller":"plugin/hclog_adapter.go:54","plugin_id":"com.mattermost.msteamsmeetings"}
{"timestamp":"2021-09-13 17:55:49.630 -04:00","level":"debug","msg":"\t/Users/dylanhaussermann/go/pkg/mod/github.com/mattermost/mattermost-server/v5@v5.28.1/plugin/client_rpc_generated.go:114 +0x78","caller":"plugin/hclog_adapter.go:54","plugin_id":"com.mattermost.msteamsmeetings"}
{"timestamp":"2021-09-13 17:55:49.630 -04:00","level":"debug","msg":"reflect.Value.call(0xc0000a3920, 0xc0000d6c60, 0x13, 0x19f0458, 0x4, 0xc000511f08, 0x3, 0x3, 0xc000334050, 0xc000061f68, ...)","caller":"plugin/hclog_adapter.go:54","plugin_id":"com.mattermost.msteamsmeetings"}
{"timestamp":"2021-09-13 17:55:49.630 -04:00","level":"debug","msg":"\t/usr/local/go/src/reflect/value.go:476 +0x8e7","caller":"plugin/hclog_adapter.go:54","plugin_id":"com.mattermost.msteamsmeetings"}
{"timestamp":"2021-09-13 17:55:49.630 -04:00","level":"debug","msg":"reflect.Value.Call(0xc0000a3920, 0xc0000d6c60, 0x13, 0xc000061f08, 0x3, 0x3, 0xc000334050, 0x18dcee0, 0xc0003218d0)","caller":"plugin/hclog_adapter.go:54","plugin_id":"com.mattermost.msteamsmeetings"}
{"timestamp":"2021-09-13 17:55:49.630 -04:00","level":"debug","msg":"\t/usr/local/go/src/reflect/value.go:337 +0xb9","caller":"plugin/hclog_adapter.go:54","plugin_id":"com.mattermost.msteamsmeetings"}
{"timestamp":"2021-09-13 17:55:49.630 -04:00","level":"debug","msg":"net/rpc.(*service).call(0xc000214880, 0xc0000a4960, 0xc0000d34b8, 0xc0000d34d0, 0xc00012c780, 0xc0000de9e0, 0x183a380, 0xc0000b0710, 0x16, 0x183a3c0, ...)","caller":"plugin/hclog_adapter.go:54","plugin_id":"com.mattermost.msteamsmeetings"}
{"timestamp":"2021-09-13 17:55:49.630 -04:00","level":"debug","msg":"\t/usr/local/go/src/net/rpc/server.go:377 +0x189","caller":"plugin/hclog_adapter.go:54","plugin_id":"com.mattermost.msteamsmeetings"}
{"timestamp":"2021-09-13 17:55:49.630 -04:00","level":"debug","msg":"created by net/rpc.(*Server).ServeCodec","caller":"plugin/hclog_adapter.go:54","plugin_id":"com.mattermost.msteamsmeetings"}
{"timestamp":"2021-09-13 17:55:49.630 -04:00","level":"debug","msg":"\t/usr/local/go/src/net/rpc/server.go:474 +0x44d","caller":"plugin/hclog_adapter.go:54","plugin_id":"com.mattermost.msteamsmeetings"}
{"timestamp":"2021-09-13 17:55:49.633 -04:00","level":"error","msg":"RPC call ExecuteCommand to plugin failed.","caller":"plugin/client_rpc_generated.go:104","plugin_id":"com.mattermost.msteamsmeetings","error":"unexpected EOF"}
{"timestamp":"2021-09-13 17:55:49.633 -04:00","level":"debug","msg":"plugin process exited","caller":"plugin/hclog_adapter.go:52","plugin_id":"com.mattermost.msteamsmeetings","wrapped_extras":"pathplugins/com.mattermost.msteamsmeetings/server/dist/plugin-darwin-amd64pid70827errorexit status 2"}
{"timestamp":"2021-09-13 17:55:49.633 -04:00","level":"error","msg":"/mstmeetings command crashed the com.mattermost.msteamsmeetings plugin. Please contact your system administrator","caller":"web/context.go:105","path":"/api/v4/commands/execute","request_id":"1xuy5cz6i3rgiqoxcepxuwh9nh","ip_addr":"76.67.110.191","user_id":"6881ppyab78wz8ogrn3ocwztur","method":"POST","err_where":"ExecutePluginCommand","http_code":500,"err_details":"err= Plugin com.mattermost.msteamsmeetings crashed due to /mstmeetings command"}
{"timestamp":"2021-09-13 17:55:49.633 -04:00","level":"debug","msg":"Received HTTP request","caller":"web/handlers.go:105","method":"POST","url":"/api/v4/commands/execute","request_id":"1xuy5cz6i3rgiqoxcepxuwh9nh","status_code":"500"}
{"timestamp":"2021-09-13 17:55:51.208 -04:00","level":"debug","msg":"Websocket request","caller":"wsapi/websocket_handler.go:25","action":"user_update_active_status"}
{"timestamp":"2021-09-13 17:56:00.147 -04:00","level":"debug","msg":"Received HTTP request","caller":"web/handlers.go:105","method":"POST","url":"/api/v4/users/status/ids","request_id":"5ygaudic4tb17mswrndzukcnfw","status_code":"200"}
{"timestamp":"2021-09-13 17:56:02.556 -04:00","level":"warn","msg":"Health check failed for plugin","caller":"plugin/health_check.go:59","id":"com.mattermost.msteamsmeetings","error":"plugin RPC connection is not responding"}
{"timestamp":"2021-09-13 17:56:02.556 -04:00","level":"debug","msg":"Deactivating plugin due to multiple crashes","caller":"plugin/health_check.go:65","id":"com.mattermost.msteamsmeetings"}
{"timestamp":"2021-09-13 17:56:02.556 -04:00","level":"error","msg":"RPC call OnDeactivate to plugin failed.","caller":"plugin/client_rpc_generated.go:33","plugin_id":"com.mattermost.msteamsmeetings","error":"connection is shut down"}
{"timestamp":"2021-09-13 17:56:02.556 -04:00","level":"warn","msg":"error closing client during Kill","caller":"plugin/hclog_adapter.go:70","plugin_id":"com.mattermost.msteamsmeetings","wrapped_extras":"errconnection is shut down"}
{"timestamp":"2021-09-13 17:56:02.556 -04:00","level":"warn","msg":"plugin failed to exit gracefully","caller":"plugin/hclog_adapter.go:72","plugin_id":"com.mattermost.msteamsmeetings"}
{"timestamp":"2021-09-13 17:56:03.743 -04:00","level":"debug","msg":"Received HTTP request","caller":"web/handlers.go:105","method":"POST","url":"/api/v4/users/status/ids","request_id":"xwxnh6ebgfn7irsrzikfq1dsyo","status_code":"200"}
{"timestamp":"2021-09-13 17:56:08.558 -04:00","level":"debug","msg":"Received HTTP request","caller":"web/handlers.go:105","method":"POST","url":"/api/v4/channels/members/me/view","request_id":"yysr51osopro7y89osnxntkzeh","status_code":"200"}
{"timestamp":"2021-09-13 17:56:16.088 -04:00","level":"debug","msg":"Received HTTP request","caller":"web/handlers.go:105","method":"POST","url":"/api/v4/users/status/ids","request_id":"47morgcbiifwmpf8tsfebq6d8o","status_code":"200"}
{"timestamp":"2021-09-13 17:56:18.520 -04:00","level":"debug","msg":"Received HTTP request","caller":"web/handlers.go:105","method":"POST","url":"/api/v4/users/status/ids","request_id":"xhayedixcpyb3ecrudi7hxa6fa","status_code":"200"}
{"timestamp":"2021-09-13 17:56:26.818 -04:00","level":"debug","msg":"Next run time for scheduler","caller":"jobs/schedulers.go:216","scheduler_name":"ResendInvitationEmailJobScheduler","next_runtime":"2021-09-13 17:56:31.81849 -0400 EDT m=+22250.599877097"}
{"timestamp":"2021-09-13 17:56:30.966 -04:00","level":"debug","msg":"Received HTTP request","caller":"web/handlers.go:105","method":"GET","url":"/api/v4/users/me/teams/f14sdjkscjdtm8hz98g6mqm1ac/channels","request_id":"5peby1cdjtgoiptga3yxsr1qzc","status_code":"304"}
{"timestamp":"2021-09-13 17:56:30.992 -04:00","level":"debug","msg":"Received HTTP request","caller":"web/handlers.go:105","method":"GET","url":"/api/v4/users/me/teams/f14sdjkscjdtm8hz98g6mqm1ac/channels/members","request_id":"65zkjtg6of8gjro9i33rmwac9a","status_code":"200"}
{"timestamp":"2021-09-13 17:56:31.042 -04:00","level":"debug","msg":"Received HTTP request","caller":"web/handlers.go:105","method":"POST","url":"/api/v4/channels/members/me/view","request_id":"4tmsb147aj8dfgzp6b47hzn9qy","status_code":"200"}
{"timestamp":"2021-09-13 17:56:32.686 -04:00","level":"debug","msg":"Received HTTP request","caller":"web/handlers.go:105","method":"POST","url":"/api/v4/users/status/ids","request_id":"1bg49m5cypbobxm7kb4p7pmrhc","status_code":"200"}
{"timestamp":"2021-09-13 17:56:39.494 -04:00","level":"debug","msg":"Received HTTP request","caller":"web/handlers.go:105","method":"POST","url":"/api/v4/users/status/ids","request_id":"xoj7perjbbf5bp8ewh91uqxd4c","status_code":"200"}
{"timestamp":"2021-09-13 17:56:50.192 -04:00","level":"debug","msg":"websocket.read: client side closed socket","caller":"app/web_conn.go:783","user_id":"6881ppyab78wz8ogrn3ocwztur"}
{"timestamp":"2021-09-13 17:56:50.193 -04:00","level":"debug","msg":"Received HTTP request","caller":"web/handlers.go:105","method":"GET","url":"/api/v4/websocket","request_id":"6pzn4sp7jiy4dxk4wbznho3z4y"}
{"timestamp":"2021-09-13 17:56:50.348 -04:00","level":"debug","msg":"Received HTTP request","caller":"web/handlers.go:105","method":"POST","url":"/api/v4/channels/members/me/view","request_id":"rt8esarg57d4pcn4gk4txe5jsa","status_code":"200"}
{"timestamp":"2021-09-13 17:56:51.243 -04:00","level":"debug","msg":"Received HTTP request","caller":"web/handlers.go:105","method":"POST","url":"/api/v4/roles/names","request_id":"hanaoags4bn4pj86mqkss9tcma","status_code":"200"}
{"timestamp":"2021-09-13 17:56:51.333 -04:00","level":"debug","msg":"Received HTTP request","caller":"web/handlers.go:105","method":"GET","url":"/api/v4/config/environment","request_id":"zf3tdm3j43bfmng4w1kuye8s1c","status_code":"200"}
{"timestamp":"2021-09-13 17:56:51.348 -04:00","level":"debug","msg":"Received HTTP request","caller":"web/handlers.go:105","method":"GET","url":"/api/v4/config","request_id":"uh8idejxttg6pbcxb4c6xaiwjh","status_code":"200"}
{"timestamp":"2021-09-13 17:56:51.439 -04:00","level":"debug","msg":"Websocket request","caller":"wsapi/websocket_handler.go:25","action":"user_update_active_status"}
{"timestamp":"2021-09-13 17:56:51.567 -04:00","level":"debug","msg":"Received HTTP request","caller":"web/handlers.go:105","method":"GET","url":"/api/v4/plugins","request_id":"oiwm5y4dnife7mgkk5asp4mjbr","status_code":"200"}
{"timestamp":"2021-09-13 17:56:51.586 -04:00","level":"debug","msg":"Received HTTP request","caller":"web/handlers.go:105","method":"GET","url":"/api/v4/analytics/old","request_id":"7fdpxefgetrhufjxad8jbmp3ew","status_code":"200"}
{"timestamp":"2021-09-13 17:56:51.665 -04:00","level":"debug","msg":"Received HTTP request","caller":"web/handlers.go:105","method":"GET","url":"/api/v4/license/client","request_id":"5hixunmmpfd3uxzuxrx4ihr77h","status_code":"200"}
{"timestamp":"2021-09-13 17:56:51.765 -04:00","level":"debug","msg":"Received HTTP request","caller":"web/handlers.go:105","method":"GET","url":"/api/v4/trial-license/prev","request_id":"5w6ypy54s7fqdgu5jj3cg4auqa","status_code":"200"}
{"timestamp":"2021-09-13 17:56:51.805 -04:00","level":"debug","msg":"Received HTTP request","caller":"web/handlers.go:105","method":"GET","url":"/api/v4/analytics/old","request_id":"6dmq6speybn6dkgpwtezmbej9y","status_code":"200"}
{"timestamp":"2021-09-13 17:56:54.341 -04:00","level":"debug","msg":"Received HTTP request","caller":"web/handlers.go:105","method":"GET","url":"/api/v4/logs","request_id":"imq8wf7ufp8rtx61648ifp737a","status_code":"200"}
{"timestamp":"2021-09-13 17:57:00.497 -04:00","level":"debug","msg":"Received HTTP request","caller":"web/handlers.go:105","method":"GET","url":"/api/v4/plugins/webapp","request_id":"cb6kkuuxkibfxyfj9oyx1oazsa","status_code":"200"}
{"timestamp":"2021-09-13 17:57:00.805 -04:00","level":"debug","msg":"Error getting bindings","caller":"app/plugin_api.go:892","plugin_id":"com.mattermost.apps","":"app_id","":"com.mattermost.zendesk","":"error","":"user is not a sysadmin: forbidden"}
{"timestamp":"2021-09-13 17:57:00.811 -04:00","level":"debug","msg":"Received HTTP request","caller":"web/handlers.go:105","method":"GET","url":"/api/v4/users/me/teams/f14sdjkscjdtm8hz98g6mqm1ac/channels","request_id":"z1dpddbp7j8o9kqs44bdiamn8e","status_code":"304"}
```

**Comment by mickmister** [comment]
@DHaussermann Tracing the code, it seems that this is the same panic that is fixed with https://github.com/mattermost/mattermost-plugin-msteams-meetings/pull/52. I've updated this branch with that change

**Review by DHaussermann — APPROVED** [review]
Tested and passed
- Ensured all new On-Prem and Cloud licenses work as expected
- Tested legacy licenses to ensure no change in behavior
- Error log message updated 
LGTM!

**Pull request merged**